### ISSUE TITLE
`public-before-private` and `static-before-instance` are mutually exclusive

### ISSUE THREAD

**Issue by thasner** [palantir/tslint#1186]
### Bug Report

- __TSLint version__: `3.7.1`
- __TypeScript version__: `1.8.7`
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
public static foo = 2;
private static bar = 3;
public x: string;
private y: string;
```

with `tslint.json`:

```json
{
    "member-ordering": [true,
      "public-before-private",
      "static-before-instance",
      "variables-before-functions"
    ],
}
```

#### Actual behavior

> Declaration of private static member variable not allowed to appear after declaration of public instance member variable

#### Expected behavior

no error

**Comment by adidahiya** [comment]
Better support for member ordering will be available in TSLint 3.10. These options, `public-before-private` and `static-before-instance`, are now deprecated and will be removed in v4 (correct me if I'm wrong here @JKillian).

**Comment by thasner** [comment]
this is awesome, thanks everyone!

i'd love to use this asap. could you release a minor version?

**Comment by JKillian** [comment]
Yep, coming today!
